compiler+runtime/src/cpp/jank/codegen/processor.cpp

```diff
@@ -196,9 +197,10 @@ namespace jank::codegen
           }
           else if constexpr(std::same_as<T, runtime::obj::character>)
           {
-            fmt::format_to(inserter,
-                           R"(jank::make_box<jank::runtime::obj::character>({}))",
-                           typed_o->data);
+            fmt::format_to(
+              inserter,
+              R"(jank::make_box<jank::runtime::obj::character>(jank::runtime::get_char_from_repr({}).unwrap()))",
```

Comment:
Why do we call `get_char_from_repr` here and not in the ctor for character? Again, think about the API design here. jank is a C++ library.

Do we want everyone who constructs a character from a string to also need to call `get_char_from_repr`? No, we just want them to give us the string and have us do the right thing.